WhisperAPI/WhisperAPI/Services/MLAPI/Facets/FilterDocuments.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using Newtonsoft.Json;
+using WhisperAPI.Models.MLAPI;
+
+namespace WhisperAPI.Services.MLAPI.Facets
+{
+    public class FilterDocuments : IFilterDocuments
+    {
+        private static readonly log4net.ILog Log = log4net.LogManager.GetLogger(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType);
+
+        private readonly string _baseAddress;
+
+        private readonly HttpClient _httpClient;
+
+        public FilterDocuments(HttpClient httpClient, string baseAddress)
+        {
+            this._httpClient = httpClient;
+            this._baseAddress = baseAddress;
+            this.InitHttpClient();
+        }
+
+        public List<string> FilterDocumentsByFacets(FilterDocumentsParameters parameters)
+        {
+            var response = this._httpClient.PostAsync("ML/Filter/Facets", CreateStringContent(parameters)).Result;
+            response.EnsureSuccessStatusCode();
+
+            return JsonConvert.DeserializeObject<List<string>>(response.Content.ReadAsStringAsync().Result);
+        }
+
+        private static StringContent CreateStringContent(FilterDocumentsParameters parameters)
```

Comment:
CreateStringContent and InitHttpClient seem to come back in most of the classes using HttpClient. Maybe we could have a base class for those or wrap the HttpClient to have those methods.